test/lib/ufe/testChildFilter.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2020 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+
+import maya.cmds as cmds
+
+from ufeTestUtils import mayaUtils
+import ufe
+
+import unittest
+
+@unittest.skipIf(os.getenv('UFE_PREVIEW_VERSION_NUM', '0000') < '2022', 'ChildFilterTestCase is only available in Maya with UFE preview version 0.2.22 and greater')
+class ChildFilterTestCase(unittest.TestCase):
+    '''Verify the ChildFilter USD implementation.
+    '''
+
+    pluginsLoaded = False
+
+    @classmethod
+    def setUpClass(cls):
+        if not cls.pluginsLoaded:
+            cls.pluginsLoaded = mayaUtils.isMayaUsdPluginLoaded()
+
+    def setUp(self):
+        ''' Called initially to set up the Maya test environment '''
+
+        # Load plugins
+        self.assertTrue(self.pluginsLoaded)
+
+        # Open ballset.ma scene in test-samples
+        mayaUtils.openGroupBallsScene()
+
+        # Clear selection to start off
+        cmds.select(clear=True)
+
+    def testFilteredChildren(self):
```

Comment:
Basically this test will load the balls scene which has 6 balls. I verify that "Props" prim has 6 children. Then I deactivate one and verify that I now have 5 children and the deactivated ball isn't one of them. Then use the filteredChildren(InactivePrims) and verify that I get 6 and that the deactivated ball is one of them.

Reply:
Nice test, thanks.